### ISSUE TITLE
Update for increased retail quest log limit (35)

### ISSUE THREAD

**Issue by ncoghlan** [jbourlakos/world-of-warcraft-simple-quest-counter#3]
On retail, the addon is still tracking against a limit of 25 rather than the increased 35 quest limit from late last year.

Fixing the fallback in https://github.com/jbourlakos/world-of-warcraft-simple-quest-counter/blob/960fad86f2d026c8917dea3b328756e7bbb75896/Quests.lua#L238 is straightforward but `_G["MAX_QUESTS"]` still comes out as 25, so the only way I could get the limit to change locally is to hardcode the result as 35 (which presumably isn't the right way to do this).

Thanks for the addon!

**Comment by Corvan79** [comment]
I have a fix for this and also upped the supported version (##interface). But I need some temporary access to create a pull request, if possible - @jbourlakos or @Hyphie24 :)

**Comment by Hyphie24** [comment]
My version is just a fork, so I can't edit accessa for the main one, but added Corvan with access to my fork. I have used SQC, but hadn't seen any issues pop up with it, and forgot about the git.
Just create a new branch with your changes and submit pull, I can then submit for main file and hopefully get it updated soon

**Comment by Hyphie24** [comment]
Update, apparently I never moved my changes to my git version, I guess I just went blank with the quest count change and changed only on my end rather than onto the git, I updated my git files to the current working version I have, Corvan if you can still do your changes and see if there is anything else that you may have updated that I overlooked, I'll then generate a pull request with the main if nothing else needs to be changed

**Comment by Corvan79** [comment]
Excellent! Thanks :) 

I've pushed one more change since Campaign quests now count towards the Quest limit. So should no longer be ignored.

**Comment by Hyphie24** [comment]
Don't think Campaign quests count, I went around Valdrakken on an alt at 70 that hadn't done anything really story related, I hit 42/35 on quest counter (with the change), went back to previous and was at 34/35, Aiding the accord was the only one not being tracked via the cap, abandoned Aiding, grabbed another quest went to 35/35, couldn't grab anything else, deleted all campaign quests and was able to regather each one. That change isn't necessary

Correction, not all campaign

**Comment by Hyphie24** [comment]
![image](https://github.com/jbourlakos/world-of-warcraft-simple-quest-counter/assets/31455641/c8df3c22-29df-4bca-8e35-ff59fed6824d)

![image](https://github.com/jbourlakos/world-of-warcraft-simple-quest-counter/assets/31455641/4b66469a-3d18-4e7f-b3f9-c4bad559a029)

may have to find the line that considers campaign quests as counting against cap vs not

3rd Update: I'm just dumbfounded some quests count, some don't, when they do it counts up, when they don't it counts up, I think the number is just going to be off by 1-3 maybe more depending on the type of quest

**Comment by Hyphie24** [comment]
> Excellent! Thanks :)
> 
> I've pushed one more change since Campaign quests now count towards the Quest limit. So should no longer be ignored.

After spending 2 hours on it, it seems it's specific quests that are "important" quests that have to be taken no matter what, it ignores quest log cap, without the need for digging through to find specifics of how they're tracked, Just going to remove the line like you have. I did change from Shadowlands level, to Dragonflight to see if that was it, but no luck. So it's counter will just be off, but it at least tracks "correctly"

**Comment by Hyphie24** [comment]
Merged into mine, submitted pull request

**Comment by Corvan79** [comment]
Yeah, it was somewhat strange. I was hitting cap at around 28 quests, and a bunch of campaign quests. When I omitted the line and counted my quests manually, it matched, so assumed all the quest campaigns counted. But alas, apparently it is not. :)

**Comment by ynazar1** [comment]
Good news! "[C_QuestLog.GetMaxNumQuestsCanAccept](https://warcraft.wiki.gg/wiki/API_C_QuestLog.GetMaxNumQuestsCanAccept)()" is the correct total (35)

I'm really struggling with how wow classifies its quests as 'real' or not for counting reasons.

There's the second returning value of [C_QuestLog.GetNumQuestLogEntries](https://warcraft.wiki.gg/wiki/API_C_QuestLog.GetNumQuestLogEntries) returns the actual counted number of quests that the character has in their log vs the total. Has anyone played with it enough to understand what the heck it counts?

**Comment by ynazar1** [comment]
For what it's worth [C_QuestLog.GetNumQuestLogEntries](https://warcraft.wiki.gg/wiki/API_C_QuestLog.GetNumQuestLogEntries) and [C_QuestLog.GetMaxNumQuestsCanAccept](https://warcraft.wiki.gg/wiki/API_C_QuestLog.GetMaxNumQuestsCanAccept) working as intended in terms of max and count...

What's broken is quest log grouped section counting for quests. I cannot understand how it counts quests towards total vs ones it ignores.

**Comment by ynazar1** [comment]
Hi folks!

Remember to update!  

SimpleQuestCounter.toc 
`## Interface: 120001, 120000` 

for the addon to load
